Allow scan using middle mouse

eyalmazuz opened the pull request:
Closes #128 

This feature adds the option to scan terms using middle mouse button.

A new option in the setting page was added to handle middle mouse scan. Since this feature only makes sense when search method is not hover, the new checkbox and the modifier settings are now bundled under a layout, and the layout will be visible only when search method is equal to modifier (similar to scan delay being only visible when search method is hover).

The new search setting page:

Hover:
![image](https://github.com/user-attachments/assets/deeb0607-2668-4cea-9b21-90b4e08ffd47)


Modifier:
![image](https://github.com/user-attachments/assets/decfb0ce-2fe1-4212-9587-273f81fc87cd)

ripose-jp commented:
This is a good change and I like the idea of not making middle mouse click and modifier settings mutually exclusive.

You do have a small misunderstanding regarding modifier settings. Modifiers should always be visible because they are used to trigger recursive searches in term glossaries and the search widget. **Search Method** only determines how subtitles should be searched. You should restore the old layout in the settings, make "Scan using middle mouse button" always visible, and make it work for both the search widget (**Tools** → **Show Search**) and term glossaries (`glossarylabel.cpp`).

ripose-jp commented on src/gui/widgets/overlay/subtitlewidget.cpp:
```diff
@@ -393,6 +396,31 @@ void SubtitleWidget::mouseMoveEvent(QMouseEvent *event)
+    }
+
+    if(m_settings.method == Settings::SearchMethod::Modifier &&
+        m_settings.middleMouseScan && event->button() == Qt::MiddleButton)
```
When if-statements span multiple lines, I like to put each condition on its own line so that it's easier to parse out.

eyalmazuz replied:
Fixed

ripose-jp commented on src/gui/widgets/overlay/subtitlewidget.cpp:
```diff
@@ -393,6 +396,31 @@ void SubtitleWidget::mouseMoveEvent(QMouseEvent *event)
+{
+    StrokeLabel::mousePressEvent(event);
+
+    if (m_settings.pauseOnHover && !m_paused)
```
This code is unnecessary here since `mouseMoveEvent` handles this.

eyalmazuz replied:
Yeah that makes sense I removed this line from the ``mousePressEvent``

eyalmazuz commented:
> You do have a small misunderstanding regarding modifier settings. Modifiers should always be visible because they are used to trigger recursive searches in term glossaries and the search widget. 

I totally forgot about recursive search, and I was the one that originally requested the feature (and use it).

> Search Method only determines how subtitles should be searched. You should restore the old layout in the settings, make "Scan using middle mouse button" always visible

I removed the visibility alteration logic from the code, so now it always shows both regardless of which search method is used.


> and make it work for both the search widget (Tools → Show Search) and term glossaries (glossarylabel.cpp).

I updated all the relevant places in the code, now middle mouse scan can be enabled inside search and glossaries

ripose-jp commented:
Thanks for fixing that stuff. I pushed to your branch with some changes. I fixed some minor things like not calling the parent of an overloaded function, misuse of Hungarian notation in some variables, and added middle mouse search to the SearchWidget. I'll merge this after CI passes.
